x/ccv/child/module.go

```diff
@@ -270,16 +279,51 @@ func (am AppModule) OnChanOpenAck(
 	ctx sdk.Context,
 	portID,
 	channelID string,
-	counterpartyVersion string,
+	counterpartyMetadata string,
 ) error {
 	// ensure parent channel has already been created
 	if parentChannel, ok := am.keeper.GetParentChannel(ctx); ok {
-		return sdkerrors.Wrapf(ccv.ErrDuplicateChannel, "parent channel: %s already established", parentChannel)
+		return sdkerrors.Wrapf(ccv.ErrDuplicateChannel,
+			"parent channel: %s already established", parentChannel)
+	}
+
+	var md parenttypes.HandshakeMetadata
+	if err := (&md).Unmarshal([]byte(counterpartyMetadata)); err != nil {
+		return sdkerrors.Wrapf(ccv.ErrInvalidHandshakeMetadata,
+			"error unmarshalling ibc-ack metadata: ", err)
+	}
+
+	if md.Version != ccv.Version {
+		return sdkerrors.Wrapf(ccv.ErrInvalidVersion,
+			"invalid counterparty version: %s, expected %s", md.Version, ccv.Version)
 	}
 
-	if counterpartyVersion != ccv.Version {
-		return sdkerrors.Wrapf(ccv.ErrInvalidVersion, "invalid counterparty version: %s, expected %s", counterpartyVersion, ccv.Version)
+	am.keeper.SetProviderFeePoolAddrStr(ctx, md.ProviderFeePoolAddr)
+
+	///////////////////////////////////////////////////
+	// Initialize distribution token transfer channel
+	// NOTE the handshake for this channel is handled by
+	//      the ibc-go/transfer module
+
+	// reuse the connection hops for this channel for the
+	// transfer channel being created.
+	connHops := am.channelKeeper.GetChannel(ctx, portID, channelID).ConnectionHops
+
+	distrTransferMsg := channeltypes.NewMsgChannelOpenInit(
+		transfertypes.PortID,
+		transfertypes.Version,
+		channeltypes.UNORDERED,
+		connHops,
+		transfertypes.PortID,
+		"", // signer unused
+	)
+
+	resp, err := am.keeper.ChannelOpenInit(ctx, distrTransferMsg)
+	if err != nil {
+		return err
 	}
+	am.keeper.SetDistributionTransmissionChannel(ctx, resp.ChannelID)
```

Comment:
One thing interesting to note is that this logic will be an issue if the transfer handshake fails at a later point. I don't think that's possible to happen currently. But worth noting

Reply:
Okay - so if I understand this correctly, the handshake could fail and we wouldn't know it here. The solution would be to verify that the transfer channel was opened successfully before using it, if it wasn't opened successfully, we could reinitiate it at that later time... Do you think we want this feature or no? Having trouble right now with verifying that the transfer channel handshake actually succeeds (see next comment).

Reply:
Added a comment to note this